numpy/lib/function_base.py

```diff
@@ -175,54 +175,102 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
             raise AttributeError(
                 'max must be larger than min in range parameter.')
 
+    # Histogram is an integer or a float array depending on the weights.
+    if weights is None:
+        ntype = int
```

Comment:
@jaimefrio - ah, I see. Actually, testing for this uncovered a bug in bincount:

```
In [1]: import numpy as np

In [2]: values = np.ones(2**32, dtype=np.intp)

In [3]: np.bincount(values)
/Volumes/Raptor/miniconda3/envs/production/bin/python.app: line 3:  7001 Segmentation fault: 11  /Volumes/Raptor/miniconda3/envs/production/python.app/Contents/MacOS/python "$@"
```

Do you see the same issue?